### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Kvolotc**

### CODE REVIEW

File: persistence/src/main/java/com/softserve/osbb/model/House.java [hunk 1]
```diff
@@ -46,26 +43,16 @@ public Integer getHouseId() {
...
-        this.street = address;
-    }
-
+    
     @Basic
-    @Column(name = "city")
-    public String getCity() {
-        return city;
-    }
+	@Column(name = "number_house")
```
Comment (KostyantynPanchenko) [inline persistence/src/main/java/com/softserve/osbb/model/House.java:46]: Use either @Basic or @Column annotation as ther have almost the same effect

File: persistence/src/main/java/com/softserve/osbb/model/User.java [hunk 2]
```diff
@@ -313,8 +304,18 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return HashCodeBuilder.reflectionHashCode(this);
     }
+    
+    @ManyToOne
+    @JoinColumn(name = "house_id")
+    public House getHouse() {
+		return house;
```
Comment (KostyantynPanchenko) [inline persistence/src/main/java/com/softserve/osbb/model/User.java:311]: Fix indentation

File: service/src/main/java/com/softserve/osbb/service/impl/HouseServiceImpl.java [hunk 3]
```diff
@@ -67,24 +67,16 @@ public House findHouseById(Integer houseId) {
...
-    @Transactional(readOnly = true, propagation = Propagation.SUPPORTS)
-    @Override
     public List<House> findAll() {
         return houseRepository.findAll();
     }
 
+    @Transactional(readOnly = true, propagation = Propagation.SUPPORTS)
+	@Override
+	public List<House> findAllByStreetId(Integer streetId) {
+
```
Comment (KostyantynPanchenko) [inline service/src/main/java/com/softserve/osbb/service/impl/HouseServiceImpl.java]: Remove empty line

File: service/src/main/java/com/softserve/osbb/service/impl/HouseServiceImpl.java [hunk 4]
```diff
@@ -128,4 +119,11 @@ public void deleteAllHouses() {
     public Page<House> getAllHousesByOsbb(Osbb osbb, Pageable pageable) {
         return houseRepository.findByOsbb(osbb, pageable);
     }
+
+    @Transactional(readOnly = true, propagation = Propagation.SUPPORTS)
+	@Override
+	public House getByNumberHouseAndStreet(Integer numberHouse, Integer streetId) {
+
```
Comment (KostyantynPanchenko) [inline service/src/main/java/com/softserve/osbb/service/impl/HouseServiceImpl.java]: Remove empty line

File: service/src/main/java/com/softserve/osbb/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -192,4 +192,11 @@ public User findUserByEmail(String email) {
     public User update(User user) {
         return userRepository.saveAndFlush(user);
     }
+
+    @Transactional(readOnly = true, propagation = Propagation.SUPPORTS)
+	@Override
+	public User getCreatorOsbb(Integer id) {
+
```
Comment (KostyantynPanchenko) [inline service/src/main/java/com/softserve/osbb/service/impl/UserServiceImpl.java]: Remove empty line

File: web/src/main/java/com/softserve/osbb/controller/HouseController.java [hunk 6]
```diff
@@ -239,6 +239,16 @@ private PageRequest buildPageRequest(PageParams pageParams) {
         }
         return response;
     }
+    
+    @RequestMapping(value = "/street/{id}", method = RequestMethod.GET)
+    public ResponseEntity<List<House>> findByStreetId(@PathVariable("id") Integer id) {
+    	 logger.info("Get all house of street: "+id);
+    	 List<House> houses = new ArrayList<>();
+    	 if((houses = houseService.findAllByStreetId(id)) == null) {
+    		 return new ResponseEntity<>(null,HttpStatus.NOT_FOUND);
```
Comment (KostyantynPanchenko) [inline web/src/main/java/com/softserve/osbb/controller/HouseController.java:248]: Fix indentation

File: web/src/main/java/com/softserve/osbb/controller/HouseController.java [hunk 7]
```diff
@@ -288,5 +298,17 @@ public boolean isApartmentNumberValid(
...
         return new ResponseEntity<>(status);
     }
+    
+    @RequestMapping(value = "/numberHouse/{numberHouse}/street/{streetId}", method = RequestMethod.GET)
+    public ResponseEntity<House> findByNumberHouseAndStreet(@PathVariable("numberHouse") Integer numberHouse,
+    		@PathVariable("streetId") Integer streetId) {
+    	 logger.info(" Get house by numberHouse " + numberHouse + " and streetId " +streetId);
+    	 House house;
+    	 if ((house = houseService.getByNumberHouseAndStreet(numberHouse, streetId)) == null ) {
+    		  return new ResponseEntity<>(null,HttpStatus.NOT_FOUND);
```
Comment (KostyantynPanchenko) [inline web/src/main/java/com/softserve/osbb/controller/HouseController.java:308]: Fix indentation

File: web/src/main/java/com/softserve/osbb/controller/UserController.java [hunk 8]
```diff
@@ -185,4 +185,15 @@ public Integer getOsbbIdByUserId(@PathVariable("id")Integer userId){
         return userResource;
     }
     
+    @RequestMapping(value = "/creator/osbb/{id}", method = RequestMethod.GET)
+    private ResponseEntity<User> getCreatorOsbb(@PathVariable("id") Integer id) {
+    	 logger.info("Get Creator");
+    	 User user;
+    	 if((user = userService.getCreatorOsbb(id)) == null) {
+    		 return new ResponseEntity<>(null,HttpStatus.NOT_FOUND);
```
Comment (KostyantynPanchenko) [inline web/src/main/java/com/softserve/osbb/controller/UserController.java:193]: Fix indentation

File: web/src/main/java/com/softserve/osbb/dto/HouseDTO.java [hunk 9]
```diff
@@ -71,14 +64,36 @@ public Integer getApartmentCount() {
...
+	}
+
+	public void setNumberHouse(Integer numberHouse) {
+		this.numberHouse = numberHouse;
+	}
+	
+	public void setOsbb(Osbb osbb) {
+		this.osbb = osbb;
+	}
+
```
Comment (KostyantynPanchenko) [inline web/src/main/java/com/softserve/osbb/dto/HouseDTO.java:78]: Remove blank ilnes and fix indentation

File: web/src/main/java/com/softserve/osbb/dto/HouseDTO.java [hunk 10]
```diff
@@ -126,21 +131,38 @@ public HouseDTOBuilder setNumberOfInhabitants(List<Apartment> apartments) {
             this.numberOfInhabitants = inhabitantsCount;
             return this;
         }
+        
+        public Street getStreet() {
```
Comment (KostyantynPanchenko) [inline web/src/main/java/com/softserve/osbb/dto/HouseDTO.java]: Lines 135 -157 --> fix indentation, remove blank lines

File: web/src/main/resources/public/src/app/registration/registration_user/registration.component.ts [hunk 11]
```diff
@@ -241,47 +245,83 @@ export class RegistrationComponent implements OnInit {
...
     selectedCity(value: any) {
         console.log(value.text);
          if(this.streets.length!=0){
             this.itemStreet.text = '';
+            this.itemHouse.text = '';
             this.streets = [];
+            this.houses = [];
+            this.isSelectedHouse = false;
         }
+                console.log(value);
```
Comment (KostyantynPanchenko) [inline web/src/main/resources/public/src/app/registration/registration_user/registration.component.ts]: delete ALL console.log() -> they are not needed in production

File: web/src/main/resources/public/src/shared/services/creatorOsbb.service.ts [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+
+    getCreatorOsbb(osbbId: number): Observable<any> {
+        let url = this.creatorURL+osbbId;
+        console.log('URL = '+url);
+        return this.http.get(url)
+            .map((res:Response)=> res.json())
+            .catch((error)=>Observable.throw(error)); 
+     }     
+}
```
Comment (KostyantynPanchenko) [inline web/src/main/resources/public/src/shared/services/creatorOsbb.service.ts]: add new line

**Review by KostyantynPanchenko — CHANGES_REQUESTED** [review]
Please provide:
- description of your pull request
- requested changes or feedback/comments

File: persistence/src/main/java/com/softserve/osbb/model/House.java (hunk 1 above)
Comment (Kvolotc) [reply persistence/src/main/java/com/softserve/osbb/model/House.java:46]: i know)

**Review by KostyantynPanchenko — APPROVED** [review]

**Pull request merged**